### PR TITLE
Bug fix/l dunc android/125 registerlifecycle not resolvable

### PR DESCRIPTION
**Pull request by LDuncAndroid**
Initially I believed that ProGuard was removing key classes but the issue I was having was due to the AAR not being a "phat" AAR. I've disabled ProGuard for release builds as I felt it's not required as the lib is open source. I've enabled publishing the AAR to maven local.

### CODE REVIEW

File: arcgis-android-toolkit/build.gradle [hunk 1]
```diff
@@ -16,10 +16,12 @@
 
 plugins {
     id 'com.android.library'
-    id 'kotlin-android-extensions'
     id 'kotlin-android'
+    id 'kotlin-android-extensions'
     id 'kotlin-kapt'
     id 'org.jetbrains.dokka-android'
+    // https://github.com/wupdigital/android-maven-publish
+    id 'digital.wup.android-maven-publish' version '3.6.2'
```
[the remark this answers is not in the document]
Comment (LDuncAndroid) [reply arcgis-android-toolkit/build.gradle]: The one you linked is abandoned and the one I'm using was top of the Google Search results: https://www.google.com/search?hl=en&q=Android%20AAR%20maven%20publishing

File: build.gradle [hunk 2]
```diff
@@ -36,6 +36,7 @@ allprojects {
         jcenter()
         maven { url 'https://esri.bintray.com/arcgis' }
         maven { url 'http://olympus.esri.com/artifactory/arcgisruntime-repo/' }
+        mavenLocal()
```
[the remark this answers is not in the document]
Comment (LDuncAndroid) [reply build.gradle]: This wouldn't affect users getting the lib from bintray, it would only affect users working on the source.

**Comment by LDuncAndroid** [comment]
Thanks for the review @gunt0001 . I've fixed those issues. Assigning back to you for review.

File: arcgis-android-toolkit/build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (gunt0001) [reply arcgis-android-toolkit/build.gradle]: Ok

**Comment by LDuncAndroid** [comment]
> @LDuncAndroid Does the project still make some use of proguard? Just asking. I see 4 references remaining:
> 
> * The `proguard-multidex-rules.pro` file.
> * Reference to `proguard-multidex-rules.pro` in `build.gradle`.
> * The `arcgis-android-toolkit/proguard-rules.pro` file.
> * The `toolkit-test-app/proguard-rules.pro` file.

Thanks for the review @alan-edi .

There should now be zero references to proguarding in the `arcgis-android-toolkit` module, which is the module that is exported as an AAR. As for proguarding the `toolkit-test-app` module, leaving an example of how to use ProGuard in the repo just helps the user when they come to using the lib and they wish to ProGuard.

**Pull request merged**